From ba2f900b2d5fbe5ef9ff75f24748b297649c7ef9 Mon Sep 17 00:00:00 2001
From: Hawking Zhang <Hawking.Zhang@amd.com>
Date: Sun, 12 Jun 2016 10:56:49 +0800
Subject: [PATCH 0563/1722] drm/amd/dal: Add ARGB and ABGR 10 bit support

The change will add ARGB and ABGR 10 bit format in to dal default rgb formats for primary plane
so that it can pass plane pixel format check when performing mode setting. In addition,
32 bit bpp/30 bit depth format is translated to DRM_FORMAT_XRGB2101010 in drm_mode_legacy_fb_format,
So add DRM_FORMAT_XRGB2101010 as the case for the correct surface format translation.

Signed-off-by: Hawking Zhang <Hawking.Zhang@amd.com>
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Reviewed-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index 2c1db40..f359fa3 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -440,9 +440,11 @@ static void fill_plane_attributes_from_fb(
 	case DRM_FORMAT_ARGB8888:
 		surface->format = SURFACE_PIXEL_FORMAT_GRPH_ARGB8888;
 		break;
+	case DRM_FORMAT_XRGB2101010:
 	case DRM_FORMAT_ARGB2101010:
 		surface->format = SURFACE_PIXEL_FORMAT_GRPH_ARGB2101010;
 		break;
+	case DRM_FORMAT_XBGR2101010:
 	case DRM_FORMAT_ABGR2101010:
 		surface->format = SURFACE_PIXEL_FORMAT_GRPH_ABGR2101010;
 		break;
@@ -1564,6 +1566,10 @@ static uint32_t rgb_formats[] = {
 	DRM_FORMAT_XRGB8888,
 	DRM_FORMAT_ARGB8888,
 	DRM_FORMAT_RGBA8888,
+	DRM_FORMAT_XRGB2101010,
+	DRM_FORMAT_XBGR2101010,
+	DRM_FORMAT_ARGB2101010,
+	DRM_FORMAT_ARGB2101010,
 };
 
 int amdgpu_dm_crtc_init(struct amdgpu_display_manager *dm,
-- 
2.7.4

